src/components/skill/skillTable.tsx

```diff
@@ -117,19 +116,20 @@ const SkillTable: React.FC = () => {
       });
     } catch (error) {
       // Revert the status change if the API call fails
-      setSkillData((prevData) =>
-        prevData.map((Skill) =>
-          Skill._id === labelId
-            ? {
-                ...Skill,
-                status: checked ? statusType.inactive : statusType.active,
-              } // revert back to original status
-            : Skill,
-        ),
-      );
+      setSkillData((prevSkillData) => {
+        // Create a shallow copy of the existing array
+        const updatedSkillData = [...prevSkillData];
+
+        updatedSkillData[index].status = checked
+          ? statusType.active
```

Comment:
This will not revert state as the true false logic is same as done in try, I think it should be opposite of what is done inside try block